src/main/java/edu/rosehulman/quota/AddUserController.java

```diff
@@ -1,4 +1,30 @@
 package edu.rosehulman.quota;
 
-public class AddUserController {  
+import spark.Request;
+import spark.Response;
+import spark.Route;
+import org.apache.http.HttpException;
+
+public class AddUserController implements Route {
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    String partnerId = request.params(":partnerId");
+    String productId = request.params(":productId");
+    String userId = request.params(":userId");
+
+    // Add the user to our database
+    boolean added = Database.getInstance().addUser(partnerId, productId, userId);
+    if (!added) {
+      throw new HttpException();
```

Comment:
You can just throw a normal exception. Saying it was an http exception when it was a database error will make it hard to debug in the future when we are looking through our error logs.

Reply:
Codacy yelled at me when I did a normal exception.

Reply:
What happens if you put a message like

```java
throw new Exception("Database lookup failed");
```